tests/load/filesystem/test_filesystem_common.py

```diff
@@ -94,3 +100,56 @@ def test_filesystem_instance_from_s3_endpoint(environment: Dict[str, str]) -> No
     assert bucket_name == "dummy-bucket"
     assert filesystem.key == "fake-access-key"
     assert filesystem.secret == "fake-secret-key"
+
+
+def test_filesystem_configuration_with_additional_arguments() -> None:
+    config = FilesystemConfiguration(
+        bucket_url="az://root", kwargs={"use_ssl": True}, client_kwargs={"verify": "public.crt"}
+    )
+    assert dict(config) == {
+        "bucket_url": "az://root",
+        "credentials": None,
+        "kwargs": {"use_ssl": True},
+        "client_kwargs": {"verify": "public.crt"},
+    }
+
+
+@pytest.mark.usefixtures("preserve_environ", "autouse_test_storage")
+def test_client_kwargs_propagate_to_instance(default_buckets_env: str) -> None:
+    config = get_config()
+    config = FilesystemConfiguration(
+        bucket_url=config.bucket_url,
+        credentials=config.credentials,
+        kwargs={"use_ssl": True},
+        client_kwargs={"verify": False, "foo": "bar"},
+    )
+
+    filesystem, _ = fsspec_from_config(config)
+
+    assert hasattr(filesystem, "client_kwargs")
+    assert ("verify", False) in filesystem.client_kwargs.items()
+    assert ("foo", "bar") in filesystem.client_kwargs.items()
+
+
+@pytest.mark.skipif("s3" not in ALL_FILESYSTEM_DRIVERS, reason="s3 destination not configured")
+def test_s3_wrong_client_certificate(default_buckets_env: str, self_signed_cert: str) -> None:
+    """Test whether filesystem raises an SSLError when trying to establish
+    a connection with the wrong client certificate."""
+    config = get_config()
```

Comment:
do not create config explicitly below. look at `test_filesystem_instance_from_s3_endpoint` and set env variables yourself. you can import `AWS_BUCKET` from `tests/load/utils.py` and set `kwargs` and `client_kwargs` in env.

this will be handy when updating the documentation

Reply:
Ahh! Thanks for the pointer, will update

Reply:
Fixed in https://github.com/dlt-hub/dlt/commit/2c6ebc2de3b0572aec025bbe35d9fa067ad8c16f